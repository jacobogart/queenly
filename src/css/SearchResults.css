.searchResults {
  background-color: #202828;
  box-shadow: #2e5657 5px 5px 5px;
  border-radius: 20px;
  position: absolute;
  z-index: 10;
  top: -110px;
  left: -30%;
  height: 100%;
  width: 150%;
  padding: 25px;
  display: flex;
  flex-wrap: wrap;
  overflow: scroll;
}
