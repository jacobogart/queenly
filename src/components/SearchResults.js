import React, { Component } from "react";
import Thumbnail from "./Thumbnail.js";

import "../css/SearchResults.css";

export default class SearchResults extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    let results;

    if (this.props.searchResults.length === 0) {
      results = (
        <h2>Please Broaden Your Search</h2>
      )
    } else {
      results = this.props.searchResults.map(result => (
        <Thumbnail
          imgURL={result.imageURL}
          selectResult={this.props.selectSearchResult}
          name={result.name}
          key={result.id}
        />
      ))
    }

    return (
      <section className="searchResults">
        <button
          className="toggle-close"
          onClick={this.props.displaySplashPage}
        >
          <i className="fas fa-times-circle" />
        </button>
        {results}
      </section>
    );
  }
}
