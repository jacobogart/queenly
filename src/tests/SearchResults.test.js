import React from "react";
import ReactDOM from "react-dom";
import SearchResults from "../components/SearchResults";
import { shallow } from "enzyme";

const mockSelectSearchResult = jest.fn();
const mockDisplaySplashPage = jest.fn();


const mockBars = [{ shows: [] }];
const mockQueens = [];
const mockSearchResults = [
  {
    bio:
      "❤️🧡💛💚💙💜💙💚💛🧡❤️ Double-D list Celebrity📍Chicago Drag Queen and Opinionated Scum ❤️🧡💛💚💙💜💙💚💛🧡❤️ IMHO EP. 4 💋⬇️",
    facebook: "https://www.facebook.com/auntycherrychan",
    id: 121,
    imageURL:
      "https://scontent-ort2-1.cdninstagram.com/vp/f0b2195af69b918c1baa1ed18c56d446/5D4AF54C/t51.2885-19/s320x320/51218190_377069173112153_1261515169956102144_n.jpg?_nc_ht=scontent-ort2-1.cdninstagram.com",
    instagram: "https://www.instagram.com/aunty_chan/?hl=en",
    name: "Aunty Chan",
    shows: ["Drag Matinee"],
    twitter: "https://twitter.com/aunty_chan?lang=en"
  }
];

describe("SearchResults", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SearchResults
        displaySplashPage={mockDisplaySplashPage}
        searchResults={mockSearchResults}
        selectSearchResult={mockSelectSearchResult}
      />
    );
  });
  it("should match snapshop", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
