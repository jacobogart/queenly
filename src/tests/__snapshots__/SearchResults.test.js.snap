// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults should match snapshop 1`] = `
<section
  className="searchResults"
>
  <button
    className="toggle-close"
    onClick={[MockFunction]}
  >
    <i
      className="fas fa-times-circle"
    />
  </button>
  <Thumbnail
    imgURL="https://scontent-ort2-1.cdninstagram.com/vp/f0b2195af69b918c1baa1ed18c56d446/5D4AF54C/t51.2885-19/s320x320/51218190_377069173112153_1261515169956102144_n.jpg?_nc_ht=scontent-ort2-1.cdninstagram.com"
    key="121"
    name="Aunty Chan"
    selectSearchResult={[MockFunction]}
  />
</section>
`;
